app/models/validators/audio_format_validator.rb

```diff
@@ -0,0 +1,42 @@
+class AudioFormatValidator < ActiveModel::EachValidator
+  FORMATS = %w(mp3 wav flac).freeze
+  BIT_RATES = [96, 112, 128, 160, 192, 224, 256, 320].freeze
+  BIT_DEPTHS = [16, 24, 32].freeze
+  CHANNELS = [1, 2].freeze
+  SAMPLE_RATES = [8000, 11025, 12000, 16000, 22050, 24000, 44100, 48000]
+
+  def validate_each(record, attribute, value)
+    return if value.nil?
+
+    # dovetail style: format, bitrate/bitdepth, channels, sample
+    unless value.is_a?(Hash) && value.keys.sort == %w(b c f s)
```

Comment:
Maybe it's weird to have the literal dovetail arrangement format in here?  But we can always convert it - validating this at least keeps things sane.